modules/average_month.py

```diff
@@ -0,0 +1,109 @@
+"""
+This module generates a report which displays average highest and lowest temperature
+as well as average max humidity for a given month
+"""
+import constants
+from modules.utils import (
+    get_highest_temperature,
+    get_lowest_temperature,
+    get_mean_humidity,
+    read_data,
+)
+
+
+def calculate_averages(pattern, path):
+    """
+    This function reads all the files at the given path which contain the given pattern
+    from each file that matches the pattern finds the following:
+    1. avg highest temperature
+    2. avg lowest temperature
+    3. avg mean humidity
+    Args:
+        pattern(str): a string which contains a year/month e.g '2006/5', '2007/3'
+        path(str): a string which contains path to a directory containing weather files
+    Returns:
+        (list or None): list of length 3, averages where
+                        averages[0] = avg_highest_temperature
+                        averages[1] = avg_lowest_temperature
+                        averages[2] = avg_mean_humidity
+                        Or
+                        None for failure
+    """
+    sum_highest_temperature = 0
+    sum_lowest_temperature = 0
+    sum_mean_humidity = 0
+    lines = []
+    for lines in read_data(pattern, path):
+        if lines is None:
+            return None
+        for line in lines:
```

Comment:
We can just continue the logic if lines exist and at the end of the function, we can just return None if nothing was returned from if above.

```suggestion
        if lines:
            for line in lines:
```

Reply:
done